src/opencl_common.c

```diff
@@ -1268,6 +1268,26 @@ static char *include_source(const char *pathname, int sequential_id, const char
 		full_path = quote_str(full_path);
 	}
 
+#ifdef __CYGWIN__
+	/*
+	 * Workaround for OpenCL kernels not found when run from Windows PowerShell.
+	 * CygWin (at least in PowerShell) maps the main JtR folder to "/".
+	 */
+	const char *sandbox_path;
+
+	sandbox_path = realpath(".", NULL);
+
+	if (sandbox_path && sandbox_path[0] == '/' && !strcmp(full_path, "/run/opencl")) {
```

Comment:
Does the check for `sandbox_path && sandbox_path[0] == '/'` serve any purpose? Isn't `realpath()` supposed to always return a full pathname, thus starting with a `/`?

Reply:
Well, if `/doc` is a full pathname, then, yes.

```
Z:\debug\doc>cd
Z:\debug\doc
```
```
..\run\john --test=0 --format=sha512crypt-opencl
Device 1: AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx
Testing: sha512crypt-opencl, crypt(3) $6$ (rounds=5000) [SHA512 OpenCL]... 
sandbox_path: /doc => ../run/opencl
```
My intention was to skip sane environments, if any.

Reply:
Anything that returns `C:\...`. Or `\`.

Reply:
So in some cases `realpath()` returns a non-absolute pathname, and in those cases we don't need the new hack, correct? However, that behavior of `realpath()` is against POSIX, which says "The realpath() function shall derive [...] an absolute pathname that names the same file, whose resolution does not involve '.', '..', or symbolic links."

Anyway, I'm OK with that.

Reply:
I agree it is useless as I see it now (I never saw something without a `/`. A full path is `/cygdrive/z/...`.